python_modules/dagster-graphql/dagster_graphql/implementation/fetch_asset_checks.py

```diff
@@ -30,21 +30,26 @@ def _fetch_asset_checks(
     asset_key: AssetKey,
     check_name: Optional[str] = None,
 ) -> GrapheneAssetChecks:
+    asset_graph = ExternalAssetGraph.from_workspace(graphene_info.context)
+
     external_asset_checks = []
     for location in graphene_info.context.code_locations:
-        # check if the code location is too old to support executing asset checks individually
-        code_location_version = (location.get_dagster_library_versions() or {}).get("dagster")
-        if code_location_version and version.parse(code_location_version) < version.parse("1.4.14"):
-            can_execute_individually = (
-                GrapheneAssetCheckCanExecuteIndividually.NEEDS_USER_CODE_UPGRADE
-            )
-        else:
-            can_execute_individually = GrapheneAssetCheckCanExecuteIndividually.CAN_EXECUTE
-
         for repository in location.get_repositories().values():
```

Comment:
we can optimize by storing external checks on the asset graph, here we're fetching them twice